cpg_workflows/stages/large_cohort.py

```diff
@@ -471,7 +499,98 @@ def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput | None:
+    """
+    Convert the site-only HT to VCF shards.
+    This is used to create the VCF fragments for the VEP stage.
+    """
+
+    def expected_outputs(self, cohort: Cohort) -> dict[str, Path]:
+        """
+        The output is a directory with the VCF fragments.
+        """
+        if ht_to_vcf_version := config_retrieve(['large_cohort', 'output_versions', 'ht_to_vcf_version'], default=None):
+            ht_to_vcf_version = slugify(ht_to_vcf_version)
+
+        ht_to_vcf_version = ht_to_vcf_version or get_workflow().output_version
+        prefix = cohort.analysis_dataset.prefix() / get_workflow().name / ht_to_vcf_version
+        return {
+            # this will be the write path for fragments of sites-only VCF (header-per-shard)
+            'hps_vcf_dir': prefix / 'site_only.vqsr.vcf.bgz',
+            # this will be the file which contains the name of all fragments (header-per-shard)
+            'hps_shard_manifest': prefix / 'site_only.vqsr.vcf.bgz' / SHARD_MANIFEST,
+        }
+
+    def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput | None:
+        from cpg_workflows.large_cohort import convert_siteonly_ht_to_vcf_shards
+
+        j = get_batch().new_job(
+            'ConvertSiteOnlyHTToVcfShards',
+            (self.get_job_attrs() or {}) | {'tool': HAIL_QUERY},
+        )
+        j.image(config_retrieve(['workflow', 'driver_image']))
+
+        j.command(
+            query_command(
+                convert_siteonly_ht_to_vcf_shards,
+                convert_siteonly_ht_to_vcf_shards.run.__name__,
+                str(inputs.as_path(cohort, LoadVqsr)),
+                str(self.expected_outputs(cohort)['hps_vcf_dir'].parent / 'repartitioned'),
+                str(self.expected_outputs(cohort)['hps_vcf_dir']),
+                setup_gcp=True,
+            ),
+        )
+
+        return self.make_outputs(cohort, data=self.expected_outputs(cohort), jobs=[j])
+
+
+@stage(required_stages=[ConvertSiteOnlyHTToVcfShards])
+class LCAnnotateFragmentedVcfWithVep(CohortStage):
+    """
+    Annotate VCF with VEP.
+    """
+
+    def expected_outputs(self, cohort: Cohort) -> Path:
+        """
+        Should this be in tmp? We'll never use it again maybe?
+        """
+        if vep_version := config_retrieve(['large_cohort', 'output_versions', 'vep'], default=None):
+            vep_version = slugify(vep_version)
+
+        vep_version = vep_version or get_workflow().output_version
+
+        return cohort.analysis_dataset.prefix() / get_workflow().name / vep_version / 'vep.ht'
+
+    def queue_jobs(self, cohort: Cohort, inputs: StageInput) -> StageOutput:
+        output = self.expected_outputs(cohort)
+        manifest_file: Path = inputs.as_path(
+            target=cohort,
+            stage=ConvertSiteOnlyHTToVcfShards,
+            key='hps_shard_manifest',
+        )
+
+        with open(manifest_file, 'r') as f:
+            manifest = f.read().strip().splitlines()
```

Comment:
Probably add a check here, as this (currently) needs to be a discontinuous workflow. i.e. the manifest file will be written by the previous stage, but needs to exist before this stage can run. This can be controlled with `last_stages`, but without specific handling it makes for an esoteric error message

[RD_combiner equivalent](https://github.com/populationgenomics/production-pipelines/blob/main/cpg_workflows/stages/rd_combiner.py#L322)